File: app/src/main/java/com/mycompany/myapp/data/api/github/model/Commit.java
package com.mycompany.myapp.data.api.github.model;

import com.google.gson.annotations.SerializedName;

public class Commit {
    public static class Author {
        @SerializedName("name") String name;
    }

    public static class CommitDetails {
        @SerializedName("message") String message;
        @SerializedName("author") Author author;
    }

    @SerializedName("commit") CommitDetails commit;

    public String getCommitMessage() {
        return commit.message;
    }

    public String getAuthor() {
        return commit.author.name;
    }
}

File: app/src/main/java/com/mycompany/myapp/data/api/github/model/Commit.kt
package com.mycompany.myapp.data.api.github.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Author(
        @Json(name = "name") val name: String)

@JsonClass(generateAdapter = true)
data class CommitDetails(
        @Json(name = "message") val message: String,
        @Json(name = "author") val author: Author)

@JsonClass(generateAdapter = true)
data class Commit(
        @Json(name = "commit") val commit: CommitDetails) {

    val commitMessage: String
        get() = commit.message

    val author: String
        get() = commit.author.name
}
